implementation/src/main/java/io/smallrye/metrics/setup/config/MetricsConfigurationManager.java

```diff
@@ -0,0 +1,320 @@
+    static final String MP_TIMER_MIN_CONFIG = "mp.metrics.distribution.timer.min-value";
+
+    private static MetricsConfigurationManager instance;
+
+    private MetricsConfigurationManager() {
+    };
+
+    private volatile Map<String, Collection<MetricPercentileConfiguration>> percentilesConfigMap = new HashMap<String, Collection<MetricPercentileConfiguration>>();
+
+    private volatile Map<String, Collection<HistogramBucketConfiguration>> histogramBucketsConfigMap = new HashMap<String, Collection<HistogramBucketConfiguration>>();;
+
+    private volatile Map<String, Collection<TimerBucketConfiguration>> timerBucketsConfigMap = new HashMap<String, Collection<TimerBucketConfiguration>>();
+
+    private volatile Map<String, Collection<DefaulBucketConfiguration>> defaultBucketConfigMap = new HashMap<String, Collection<DefaulBucketConfiguration>>();
+
+    private volatile Map<String, Collection<HistogramBucketMaxConfiguration>> defaultHistogramBucketMaxConfig = new HashMap<String, Collection<HistogramBucketMaxConfiguration>>();
+    private volatile Map<String, Collection<HistogramBucketMinConfiguration>> defaultHistogramBucketMinConfig = new HashMap<String, Collection<HistogramBucketMinConfiguration>>();
+
+    private volatile Map<String, Collection<TimerBucketMaxConfiguration>> defaultTimerBucketMaxConfig = new HashMap<String, Collection<TimerBucketMaxConfiguration>>();
+    private volatile Map<String, Collection<TimerBucketMinConfiguration>> defaultTimerBucketMinConfig = new HashMap<String, Collection<TimerBucketMinConfiguration>>();
+
+    public static synchronized MetricsConfigurationManager getInstance() {
+        if (instance == null) {
+            instance = new MetricsConfigurationManager();
+        }
+        return instance;
+    }
+
+    /**
+     * Returns the matching configuration object if it exists, null otherwise
+     * 
+     * @param metricName the metric name to check configuration against
+     * @return the matching configuration object if it exists, null otherwise
+     */
+    public synchronized MetricPercentileConfiguration getPercentilesConfiguration(String metricName) {
+
+        String appName = getApplicationName();
+
+        Collection<MetricPercentileConfiguration> computedValues = percentilesConfigMap.computeIfAbsent(appName, f -> {
+            Optional<String> input = ConfigProvider.getConfig().getOptionalValue(MP_PERCENTILES_PROP, String.class);
+
+            return (input.isPresent()) ? MetricPercentileConfiguration.parseMetricPercentiles(input.get()) : null;
+
+        });
+
+        if (computedValues != null && computedValues.size() != 0) {
+            MetricPercentileConfiguration retVal = MetricPercentileConfiguration.matches(computedValues, metricName);
+            if (LOGGER.isLoggable(Level.FINEST)) {
+                LOGGER.logp(Level.FINEST, CLASS_NAME, null,
+                        "Returning \"{0}\" configuration for metric:\"{0}\" with values: {1} ",
```

Comment:
Should the parameter be {0}, {1}, and {2}?

Reply:
xfixed